### PR TITLE
Do not expose unsupported queries in the GraphQL schema.

### LINKED ISSUE #16008: Expose SQL Queries in GraphQL too
### Is your feature request related to a problem? Please describe.
Related: https://github.com/OrchardCMS/OrchardCore/discussions/15959. Lucene and Elasticsearch Queries are accessible from GraphQL, see [here](https://docs.orchardcore.net/en/main/reference/modules/Queries/#graphql). This makes it possible to expose arbitrary content lists in GraphQL. However, the same feature is not available for SQL Queries.

### Describe the solution you'd like
Have the same kind of GraphQL integration for SQL Queries that we have for the other two.

### Describe alternatives you've considered
Alternatively, we could somehow expose field querying with Content Fields Indexing (SQL) in GraphQL, also see https://github.com/OrchardCMS/OrchardCore/discussions/15959#discussioncomment-9322540 and https://github.com/OrchardCMS/OrchardCore/issues/7047, https://github.com/OrchardCMS/OrchardCore/issues/12781 about this.

### PR DESCRIPTION
**Pull request by gvkries**
Fixes #16008

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Search.Lucene/GraphQL/LuceneQueryFieldTypeProvider.cs [hunk 1]
```diff
@@ -85,7 +85,7 @@ public async Task BuildAsync(ISchema schema)
 
         private static FieldType BuildSchemaBasedFieldType(LuceneQuery query, JsonNode querySchema, string fieldTypeName)
         {
-            var properties = querySchema["properties"].AsObject(); ;
+            var properties = querySchema["properties"]?.AsObject(); ;
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Search.Lucene/GraphQL/LuceneQueryFieldTypeProvider.cs:88]:
```suggestion
            var properties = querySchema["properties"]?.AsObject(); 
```

**Comment by hishamco** [comment]
@hyzx86 do you have any comments on this PR?

File: src/OrchardCore.Modules/OrchardCore.Queries/Sql/GraphQL/SqlQueryFieldTypeProvider.cs [hunk 2]
```diff
@@ -62,7 +62,8 @@ public async Task BuildAsync(ISchema schema)
 
                     var fieldTypeName = querySchema["fieldTypeName"]?.ToString() ?? query.Name;
 
-                    if (type.StartsWith("ContentItem/", StringComparison.OrdinalIgnoreCase))
+                    if (query.ReturnDocuments &&
+                        type.StartsWith("ContentItem/", StringComparison.OrdinalIgnoreCase))
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Queries/Sql/GraphQL/SqlQueryFieldTypeProvider.cs:66]:
```suggestion
                    if (query.ReturnDocuments && type.StartsWith("ContentItem/", StringComparison.OrdinalIgnoreCase))
```

**Review by hishamco — APPROVED** [review]

**Review by hishamco — APPROVED** [review]

**Comment by hishamco** [comment]
@gvkries is there anything else here or shall I merge?

**Comment by gvkries** [comment]
This can be merged.

**Pull request merged**